lgc/include/lgc/util/Internal.h

```diff
@@ -96,4 +96,8 @@ bool isDontCareValue(llvm::Value *value);
 // type in a return value struct, ensuring it gets into VGPRs.
 llvm::Type *getVgprTy(llvm::Type *ty);
 
+// Modify the function argument types, and return the new function.
+llvm::Function *mutateFunctionArguments(llvm::Function &fn, llvm::Type *retTy,
+                                        const llvm::ArrayRef<llvm::Type *> argTys, llvm::AttributeList attributes);
+
```

Comment:
Maybe integrate the argument fixup into this utility? We could pass an additional list of indices, mapping between old and new arguments.

Otherwise, we should clearly mention here that the user needs to fix up arguments.

Reply:
I would rather keep the uses replacement out of this helper function for simplicity. Caller usually need to do some more work on the arguments. So they can just handle the replacement with other work in one loop.

Reply:
Btw, I have updated the comments per your suggestion.